pkg/controller/operconfig/migration.go

```diff
@@ -97,3 +120,325 @@ func convertEgressFirewallToEgressNetworkPolicy(ctx context.Context, client cnoc
+	if err != nil {
+		return false, err
+	}
+
+	for _, node := range nodeList {
+		found := false
+		for _, hostSubnet := range hostSubnetList {
+			if hostSubnet.Object["host"] == node.Name {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false, nil
+		}
+	}
+
+	result, err := netNamespacesExistForAllNamespaces(ctx, client)
+	if !result {
+		return false, nil
+	} else if err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+func netNamespacesExistForAllNamespaces(ctx context.Context, client cnoclient.Client) (bool, error) {
+	// get all namespaces
+	// get all netnamespaces
+	// iterate over all namespaces and return false if any of them don't have an associated netnamespace
+	namespaceList, err := ListAllNamespaces(ctx, client)
+	if err != nil {
+		return false, err
+	}
+	netnamespaceList, err := ListAllOfSpecifiedType(gvrNetnamespace, ctx, client)
+	if err != nil {
+		return false, err
+	}
+
+	for _, namespace := range namespaceList {
+		found := false
+		for _, netnamespace := range netnamespaceList {
+			if netnamespace.Object["netname"] == namespace.Name {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false, nil
+		}
+	}
+
+	return true, nil
+}
+
+func netNamespaceHasEgressIpConfig(nns uns.Unstructured) bool {
+	if nns.Object["egressIPs"] != nil {
+		return len(nns.Object["egressIPs"].([]interface{})) > 0
+	}
+	return false
+}
+
+func hostSubnetHasEgressIpConfigAutomatic(hsn uns.Unstructured) bool {
+	if hsn.Object["egressIPs"] != nil && hsn.Object["egressCIDRs"] != nil {
+		return len(hsn.Object["egressIPs"].([]interface{})) > 0 && len(hsn.Object["egressCIDRs"].([]interface{})) > 0
+	}
+	return false
+}
+
+func hostSubnetHasEgressIpConfigManual(hsn uns.Unstructured) bool {
+	if hsn.Object["egressIPs"] != nil && hsn.Object["egressCIDRs"] == nil {
+		return len(hsn.Object["egressIPs"].([]interface{})) > 0
+	}
+	return false
+}
+
+func labelNodeAndRemoveHostSubnetConfig(ctx context.Context, client cnoclient.Client, hsn uns.Unstructured) error {
+	// label node as egressassignable
+	hostStr := fmt.Sprintf("%v", hsn.Object["host"])
+	nodeObj, err := client.Default().Kubernetes().CoreV1().Nodes().Get(ctx, hostStr, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	if _, ok := nodeObj.Labels["k8s.ovn.org/egress-assignable"]; !ok {
+		nodeObj.Labels["k8s.ovn.org/egress-assignable"] = ""
+		if err := apply.ApplyObject(ctx, client, nodeObj, ""); err != nil {
+			return err
+		}
+	}
+
+	// copy egressCIDRs to node label if possible
+	// - if egressCIDRs contains values (automatic config), on rollback this label will provide the respective hostsubnet with its original egressCIDRs values
+	// - if egressCIDRs is empty (manual config), on rollback this label will not exist, and so we default to node's subnet for egressCIDR field.
+	egressCIDRs, ok := hsn.Object["egressCIDRs"]
+	if ok {
+		// automatic configuration
+		if _, ok := nodeObj.Annotations["sdn.to.ovn.migration/sdnEgressCIDRs"]; !ok {
```

Comment:
I will make the annotation key names constant.

I propose instead this key name `networkoperator.openshift.io/sdn-to-ovn-migration-egressCIDRs`. I don't really see a reason to generalize the key name fully encoded in the annotation, since egressCIDRs are the only thing we need to preserve/restore anyway. 

We do not save egressIPs to restore them because it is for manual SDN config, which we do not support.

If we are rolling back to an automatic SDN config, egressIPs will be restored automatically by SDN once netnamespace and hostsubnet get configured with their old egressIPs and egressCIDRs respectively (this is currently implemented).

Reply:
Shouldn't this be as easy as defining a struct
```
type OVNMigrationNodeAnnotation {
    egressCIDRs []string
}
```
and then just un/marshaling this struct into/from the annotation?
And then we don't have to worry about assumptions on what might come in the future that will basically not play nice with the annotation key `sdn-to-ovn-migration-egressCIDRs` because it is too specific?

Reply:
I think because it _is_ so specific, it will actually play nice with code that comes in the future. I'm not sure what assumptions we are supposed to worry about here. However, I don't feel strongly one way or the other on this issue, so I went ahead and followed your suggestion.

Reply:
My assumption was that if we were to need additional information saved in the future, I would it is better to have 1 annotation rather than n annotations.